ietf112-slides/ppm-ietf-112.tex

```diff
@@ -0,0 +1,455 @@
+    \begin{itemize}
+    \item DNS requests
+    \item Safe Browsing queries
+    \end{itemize}
+  \end{itemize}
+\end{frame}
+
+
+\begin{frame}{Bad Use Cases for Anonymization}
+
+  \begin{itemize}
+  \item High dimensionality data (statistical queries)
+    \begin{itemize}
+    \item Multiple variables that need to be reported together
+    \item When you want to look at subgroups
+    \item Any time you want to do correlation/regression
+    \item \emph{Anonymized data needs to be disaggregated to prevent de-anonymization}
+    \end{itemize}
+
+  \item Collecting common values (heavy hitters)
+    \begin{itemize}
+    \item The ``top N'' values common $>t$ users
+    \item \emph{Anonymized data collects every value and depends on reporting only common values}      
+    \end{itemize}
+  \end{itemize}
+\end{frame}
+
+
+\begin{frame}{Cryptography to the Rescue}
+{\center
+\includegraphics[width=4in]{prio.png}  
+}
+Solutions to both statistical queries and heavy hitters share a common framework:
+\begin{itemize}
+\item Split data between two servers
+\item Each server computes aggregated shares
+\item Aggregate shares combined to produce final value  
+\end{itemize}
+\end{frame}
+
+\begin{frame}{Example: Prio~\cite{201553}}
+
+  \begin{itemize}
+  \item Useful for computing numeric aggregates (sum, mean, etc.)
+  \item[]
+  \item Each client $i$ holds a value $x_i$, then secret shares it with each server
+    \begin{itemize}
+    \item Generates random $R_i \leftarrow \mathbb{F}_p$
+    \item Sends $x_i - R_i (mod p)$ to server 1
+    \item Sends $R_i$ to server 2
+    \end{itemize}
+
+  \item Each server adds up their shares
+    \begin{itemize}
+    \item Server 1: $\sum_i x_i - R_i$
+    \item Server 2: $\sum_i R_i$
+    \end{itemize}
+
+  \item Now add these up: $\sum_i x_i - R_i + \sum_i R_i = \sum_i x_i + \sum_i R_i - \sum_i R_i = \sum_i x_i$
+  \end{itemize}
+\end{frame}
+
+
+\begin{frame}{What else can Prio compute?}
+
+  \begin{tabular}{l l}
+    Arithmetic mean & $\sum_i x_i / i$ \\
+    Product & $exp(\sum_i log(x_i))$  \\
+    Geometric mean & From product \\
+    Variance and stddeviation & From $\sum_i x_i$ and $\sum_i (x_i)^2$ \\
+    Boolean OR, AND & ... \\
+    MIN, MAX & ... \\
+    Ordinarily least squares (OLS) & ... \\
+  \end{tabular}
+
+  \vspace{2ex}
+  \begin{itemize}
+    \item[] The trick is finding the right encoding    
+    \end{itemize}
+\end{frame}
+
+\begin{frame}{What about bogus data?}
+
+  \begin{itemize}
+  \item Plausible but false
+    \begin{itemize}
+    \item ``I am 180cm tall'' when I am actually 175cm
+    \item A problem with any surveying technique
+    \item Solution: live with somewhat noisy data
+    \end{itemize}
+  \item Completely ridiculous
+    \begin{itemize}
+    \item ``I am 1km tall'' (or worse, ``I am -1km tall'')
+    \item Easy to remove with standard systems by filtering
+      \begin{itemize}
+      \item ... but with Prio the data is encrypted
```

Comment:
"encrypted" isn't the right word here, especially since the secret shares constructed by clients *do* get encrypted, but not to conceal them from aggregators. I can't come up with a better single word to express this, though.